deploy.py

```diff
@@ -0,0 +1,117 @@
+                             'this will re-config a existing OpenLab environment to '
+                             'switch the "master/slave" role of hosts',
+                        )
+    parser.add_argument('--new-slave',
+                        action='store_true',
+                        help='deploy the slave part of a CI cluster freshly, you need '
+                             'to specify the new slave hosts IP addresses with "--new-ip" '
+                             'argument, for example: '
+                             '--new-ip zuul02=192.168.5.5 --new-ip nodepool02=192.168.6.6',
+                        )
+    parser.add_argument('-e', '--extra-vars',
+                        metavar="<key=value>",
+                        action='append',
+                        default=[],
+                        help='extra vars passed to Ansible command',
+                        )
+    parser.add_argument('--new-ip',
+                        metavar="<key=value>",
+                        action='append',
+                        help='specify new ip address for current inventory, this argument '
+                             'need to use with key-value pairs combined with "=", the key '
+                             'must be the host name defined in the inventory yaml files, '
+                             'e.g. zuul01, nodepool02, allinone01.',
+                        )
+    parser.add_argument('--with-vars',
+                        action='store_true',
+                        help='show the hosts graph of ansible inventory with vars, specify with '
+                             'key-value pairs combined with "="',
+                        )
+    parser.add_argument('--graph',
+                        action='store_true',
+                        help='show the hosts graph of ansible inventory',
+                        )
+    parser.add_argument('--show-old-ips',
+                        action='store_true',
+                        help='show the old hosts ip address',
+                        )
+    parser.add_argument('--show-new-ips',
+                        action='store_true',
+                        help='show the new hosts ip address',
+                        )
+    return parser
+
+
+def main():
+    parsed_args = add_cli_args().parse_args()
+    os.environ['OL_TYPE'] = parsed_args.type
+    cmd = ['ansible-playbook', '-i', 'inventory/inventory.py', 'playbooks/site.yaml']
+
+    if parsed_args.user:
+        cmd += ['-u', parsed_args.user]
+
+    if parsed_args.new_ip:
+        specified_ips = dict([(d.partition('=')[::2]) for d in parsed_args.new_ip])
+        for host, ip in specified_ips.items():
+            os.environ['OL_%s_IP' % host.upper()] = ip
+
+    if parsed_args.new_slave:
+        if not parsed_args.new_ip:
+            sys.exit("ERROR: Must specify the new slave hosts IPs with --new-ip")
+        os.environ['OL_SWITCH_MASTER_SLAVE'] = True
```

Comment:
the `OL_SWITCH_MASTER_SLAVE=True` is used by the dynamic inventory script `inventory/inventory.py`, it is used to automatically config the old `slave` IPs to the new `master` IPs, and user only need to specify the new slave IPs, otherwise, user need to set 4 IPs than 2 IPs by `--new-ip`

Reply:
Thank your clarification